enterprise/server/remote_execution/action_merger/action_merger.go

```diff
@@ -282,6 +282,11 @@ func DeletePendingExecution(ctx context.Context, rdb redis.UniversalClient, exec
 		return nil
 	}
 
+	// The pending execution should be cleaned up regardless of whether
+	// the original request is canceled or times out.
+	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), 5*time.Second)
+	defer cancel()
```

Comment:
there is another caller that is already extending the timeout by 10s ([here](https://github.com/buildbuddy-io/buildbuddy/blob/849bb7efd019b5ecbce0be5088c4e79473a51ccd/enterprise/server/remote_execution/execution_server/execution_server.go#L694)) I think for consistency, this should be done in `deletePendingExecutionOnce` ?